### PR TITLE
Update type hints to work with numpy 1.20

### PR DESCRIPTION
**Pull request by jklaise**
Same story as https://github.com/SeldonIO/alibi/pull/361.

### CODE REVIEW

File: alibi_detect/cd/base.py [hunk 1]
```diff
@@ -178,7 +178,9 @@ def predict(self, X: Union[np.ndarray, list], drift_type: str = 'batch',
          threshold after multivariate correction if needed and test statistics.
         """
         # compute drift scores
-        p_vals, dist = self.score(X)
+        # TODO: (Arnaud) should `score` take Union[np.ndarray, list] ?
```
Comment (arnaudvl) [inline alibi_detect/cd/base.py:181]: Can be just `np.ndarray`

File: alibi_detect/utils/statstest.py [hunk 2]
```diff
@@ -55,11 +56,13 @@ def permutation_test(x: Union[np.ndarray, da.array],
             dist_permutation = dist_permutation.compute()
         dist_permutations[_] = dist_permutation
         k += dist <= dist_permutation
-    outputs = (k / n_permutations,)
+    outputs = (k / n_permutations,)  # type: Any
+    # TODO: the following logic seems pretty brittle, i.e. not clear what you get in each position of the tuple
+    # should we just replace it with a dictionary or a namedtuple?
```
Comment (arnaudvl) [inline alibi_detect/utils/statstest.py:61]: WIll be taken care of in the MMDDrift PR using torch

File: alibi_detect/utils/perturbation.py [hunk 3]
```diff
@@ -633,7 +634,7 @@ def glass_blur(x: np.ndarray, sigma: float, max_delta: int, iterations: int, xra
     if xrange[0] != 0 or xrange[1] != 255:
         x = (x - xrange[0]) / (xrange[1] - xrange[0]) * 255
 
-    x = np.uint8(gaussian(x, sigma=sigma, multichannel=True))
+    x = gaussian(x, sigma=sigma, multichannel=True).astype(np.uint8)  # TODO: (Arnaud) check this works as intended
```
Comment (arnaudvl) [inline alibi_detect/utils/perturbation.py:637]: Looks good

**Comment by jklaise** [comment]
Since this is fairly out of date now I will close in favour of a new PR.

**Pull request closed without merging**